internal/controller/nonadminbackup_controller.go

```diff
@@ -128,125 +122,135 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+				logger.Error(updateErr, conditionUpdateError)
+				return true, false, updateErr
+			}
 		}
 
-		// We do not requeue - this was an error from getting Spec from NAB
-		return true, false, err
+		return true, false, reconcile.TerminalError(err)
 	}
 
-	updatedStatus, errUpdateStatus := function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, nab, nacv1alpha1.NonAdminConditionAccepted, metav1.ConditionTrue, "BackupAccepted", "backup accepted")
-	if errUpdateStatus != nil {
-		logger.Error(errUpdateStatus, "Unable to set BackupAccepted Condition: True", nameField, nab.Name, constant.NameSpaceString, nab.Namespace)
-		return true, false, errUpdateStatus
-	} else if updatedStatus {
-		// We do requeue - The VeleroBackup got accepted and next reconcile loop will continue
-		// with further work on the VeleroBackup such as creating it
+	updated := meta.SetStatusCondition(&nab.Status.Conditions,
+		metav1.Condition{
+			Type:    string(nacv1alpha1.NonAdminConditionAccepted),
+			Status:  metav1.ConditionTrue,
+			Reason:  "BackupAccepted",
+			Message: "Backup accepted",
+		},
+	)
+	if updated {
+		if err := r.Status().Update(ctx, nab); err != nil {
+			logger.Error(err, conditionUpdateError)
+			return true, false, err
+		}
+
+		logger.V(1).Info(conditionUpdateRequeue)
 		return false, true, nil
 	}
 
+	logger.V(1).Info("NonAdminBackup Spec already validated")
 	return false, false, nil
 }
 
-// CreateVeleroBackupSpec creates or updates a Velero Backup object based on the provided NonAdminBackup object.
+// UpdateSpecStatus updates the Spec and Status from the NonAdminBackup.
 //
 // Parameters:
 //
 //	ctx: Context for the request.
 //	log: Logger instance for logging messages.
 //	nab: Pointer to the NonAdminBackup object.
 //
-// The function generates a name for the Velero Backup object based on the provided namespace and name.
-// It then checks if a Velero Backup object with that name already exists. If it does not exist, it creates a new one.
+// The function generates the name for the Velero Backup object based on the provided namespace and name.
+// It then checks if a Velero Backup object with that name already exists. If it does not exist, it creates a new one
+// and updates NonAdminBackup Status. Otherwise, updates NonAdminBackup VeleroBackup Spec and Status based on Velero Backup object Spec and Status.
 // The function returns boolean values indicating whether the reconciliation loop should exit or requeue
-func (r *NonAdminBackupReconciler) CreateVeleroBackupSpec(ctx context.Context, logrLogger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (exitReconcile bool, requeueReconcile bool, errorReconcile error) {
-	logger := logrLogger.WithValues("CreateVeleroBackupSpec", nab.Namespace)
+func (r *NonAdminBackupReconciler) UpdateSpecStatus(ctx context.Context, logrLogger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (exitReconcile bool, requeueReconcile bool, errorReconcile error) {
```

Comment:
need to undo some refactor here, can not do CREATE and UPDATE call in same reconcile